crates/api_models/src/payments.rs

```diff
@@ -518,6 +518,62 @@ pub struct PaymentsRequest {
     pub merchant_order_reference_id: Option<String>,
 }
 
+/// Checks if the inner values of two options are equal
+/// Returns true if values are not equal, if values match returns false
+fn are_optional_values_invalid<T: PartialEq>(
+    first_option: Option<&T>,
+    second_option: Option<&T>,
+) -> bool {
+    first_option
+        .zip(second_option)
+        .map(|(value1, value2)| value1 != value2)
+        .unwrap_or(true)
+}
```

Comment:
This function is supposed to return `true` only when two options are Some and the inner value is different right?

But if one or more options are None, this will return `true` as well.